builders/interact/ethereum-api/dev-env/foundry.md

```diff
@@ -0,0 +1,359 @@
+
+To fork the demo EVM ContainerChain from the command line, you can run the following command from within your Foundry project directory. You can also replace the RPC URL with the RPC URL of your EVM ContainerChain.
+
+```bash
+    anvil --fork-url https://fraa-dancebox-3001-rpc.a.dancebox.tanssi.network
+```
+
+Your forked instance will have 10 development accounts that are pre-funded with 10,000 test tokens. The forked instance is available at `http://127.0.0.1:8545/`. The output in your terminal should resemble the following:
+
+![Forking terminal screen](/images/builders/interact/ethereum-api/dev-environments/foundry/foundry-5.png)
+
+To verify you have forked the network, you can query the latest block number and compare it to the current block number of the [demo EVM ContainerChain](https://tanssi-evmexplorer.netlify.app/){target=_blank}
+
+```bash
+curl --data '{"method":"eth_blockNumber","params":[],"id":1,"jsonrpc":"2.0"}' -H "Content-Type: application/json" -X POST localhost:8545 
+```
+
+If you convert the `result` from [hex to decimal](https://www.rapidtables.com/convert/number/hex-to-decimal.html){target=_blank}, you should get the latest block number from the time you forked the network. 
+
+From here you can deploy new contracts to your forked instance of the demo EVM ContainerChain (or any other EVM ContainerChain) or interact with contracts already deployed. Building off of the previous example in this guide, you can make a call using Cast to check the balance of the minted MYTOK tokens in the account you deployed the contract with:
+
+```bash
+cast call INSERT_CONTRACT_ADDRESS  "balanceOf(address)(uint256)" INSERT_YOUR_ADDRESS --rpc-url http://localhost:8545
+```
+
+## Using Chisel {: #using-chisel }
+
+Chisel is a Solidity REPL, or shell. It allows a developer to write Solidity directly in the console for testing small snippets of code, letting developers skip the project setup and contract deployment steps for what should be a quick process.  
+
+Since Chisel is mainly useful for quick testing, it can be used outside of a Foundry project. But, if executed within a Foundry project, it will keep the configurations within `foundry.toml` when running.  
+
+For this example, you will be testing out some of the features of `abi` within Solidity because it is complex enough to demonstrate how Chisel could be useful. To get started using Chisel, run the following in the command line to start the shell:
+
+```bash
+chisel
+```
+
+In the shell, you can write Solidity code as if it was running within a function:
+
+```solidity
+bytes memory myData = abi.encode(100, true, "Build with Tanssi");
+```
+
+Let's say you were interested in how `abi` encoded data, because you're looking into how to most efficiently store data on the blockchain and thus save gas. To view how the `myData` is stored in memory, you can use the following command while in the Chisel shell:  
```

Comment:
```suggestion
Let's say you were interested in how `abi` encoded data because you're looking into how to most efficiently store data on the blockchain and thus save gas. To view how the `myData` is stored in memory, you can use the following command while in the Chisel shell:  
```